.pre-commit-config.yaml

```diff
@@ -22,21 +22,22 @@ repos:
     hooks:
       - id: mypy
         args:
-          - --follow-imports=skip
-        files: | # incremental 'fully-typed' list
-          (?x)^(
-              pdm\/__init__\.py|
-              pdm\/__main__\.py|
-              pdm\/exceptions\.py|
-              pdm\/formats\/.*\.py|
-              pdm\/installers\/.*\.py|
-              pdm\/models\/auth\.py|
-              pdm\/models\/caches\.py|
-              pdm\/models\/python\.py|
-              pdm\/models\/setup\.py|
-              pdm\/models\/specifiers\.py|
-              pdm\/models\/versions\.py|
-              pdm\/project\/config\.py|
-              pdm\/termui\.py|
-              pdm\/utils\.py|
-          )$
+          - pdm
+          - |
+            --exclude=(?x)^(
+              (?!pdm\/__init__\.py)
+              (?!pdm\/__main__\.py)
+              (?!pdm\/exceptions\.py)
+              (?!pdm\/formats\/.*\.py)
+              (?!pdm\/installers\/.*\.py)
+              (?!pdm\/models\/auth\.py)
+              (?!pdm\/models\/caches\.py)
+              (?!pdm\/models\/python\.py)
+              (?!pdm\/models\/setup\.py)
+              (?!pdm\/models\/specifiers\.py)
+              (?!pdm\/models\/versions\.py)
+              (?!pdm\/project\/config\.py)
+              (?!pdm\/termui\.py)
+              (?!pdm\/utils\.py)
+            ).*\.py$ # incremental 'fully-typed' list
+        pass_filenames: false
```

Comment:
what's the benefit of inverting the logic in this way?

Reply:
https://github.com/pdm-project/pdm/pull/452/files#diff-63a9c44a44acf85fea213a857769990937107cf072831e1a26808cfde9d096b9 the latest change is here.

We want to executing `mypy` via pre-commit is same as executing `mypy` via cli directly (except the first one with different `--exclude=...`. In pre-commit we need to exclude no-fully typed files because we want to run mypy on fully typed files only. And using `--exclude=...` is easier rather than configuring them by using `[mypy-*]` in the setup.cfg file to exclude all the no-fully typed files.

> The document of `--exclude` optional argument : https://mypy.readthedocs.io/en/stable/command_line.html#cmdoption-mypy-exclude